cl/users/models.py

```diff
@@ -190,8 +259,211 @@ def generate_recap_email(user_profile: UserProfile, append: int = None) -> str:
     return recap_email
 
 
+def handle_hard_bounce(
+    message_id: str, bounceSubType: str, recipient_emails: List[str]
+) -> None:
+    """Handle a hard bounce notification received from SNS
+    :param message_id: The unique message id assigned by Amazon SES
+    :param bounceSubType: The subtype of the bounce, as determined
+     by Amazon SES
+    :param recipient_emails: a list of email addresses one per recipient
+     to whom the bounce notification pertains
+    :return: None
+    """
+    unexpected_events = ["Suppressed", "OnAccountSuppressionList"]
+    for email in recipient_emails:
+        if bounceSubType in unexpected_events:
+            # Handle unexpected bounceSubType events, log a warning
+            logging.warning(
+                f"Unexpected {bounceSubType} hard bounce for {email}"
+            )
+            # After log the event ban the email address
+            # Only ban email address if it hasn't been previously banned
+            EmailFlag.objects.get_or_create(
+                email_address=email,
+                flag_type="ban",
+                defaults={"reason": bounceSubType},
+            )
+        else:
+            # Any other hard bounce even only ban the email address
+            # Only ban email address if it hasn't been previously banned
+            EmailFlag.objects.get_or_create(
+                email_address=email,
+                flag_type="ban",
+                defaults={"reason": bounceSubType},
+            )
+
+
+def handle_soft_bounce(
+    message_id: str, bounceSubType: str, recipient_emails: List[str]
+) -> None:
+    """Handle a soft bounce notification received from SNS
```

Comment:
Should have a newline here:
```suggestion
    """Handle a soft bounce notification received from SNS
    
```